### PR TITLE
PR Auto labeler

### PR DESCRIPTION
**Pull request by snazy**
A little bit more safety when opening PRs.

See https://github.com/actions/labeler

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+pr-native:
+- 'gradle/wrapper/**'
+- 'servers/**'
+- '.github/**'
+
+pr-integrations:
+- 'gradle/wrapper/**'
+- 'clients/**'
+- '.github/**'
```
Comment (XN137) [inline .github/labeler.yml:32]: could these be more specific i.e. `.github/workflows/pull-request-helm.yml` or is there a good reason why we want to add all these labels whenever any workflow changes?

File: .github/workflows/pull-request-labeler.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  triage:
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/labeler@v4
+        with:
+          repo-token: "${{ secrets.CI_REPORTS_TOKEN }}"
```
Comment (XN137) [inline .github/workflows/pull-request-labeler.yml:29]: should we set `sync-labels: true` ?

```
Whether or not to remove labels when matching files are reverted or no longer changed by the PR
```

iceberg is using it:
https://github.com/apache/iceberg/blob/master/.github/workflows/labeler.yml

File: .github/workflows/pull-request-labeler.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Auto Labeler
+
+on:
+  pull_request_target:
```
Comment (XN137) [inline .github/workflows/pull-request-labeler.yml:18]: trailing `.` looks odd, should we use the suggestion from the action README?
```
on:
- pull_request_target
```

**Review by XN137 — APPROVED** [review]
good idea!
change seems OK as is but please consider the suggestions/questions

File: .github/labeler.yml (hunk 1 above)
Comment (snazy) [reply .github/labeler.yml:32]: TL;DR easier to reason about

There are also custom actions and such

File: .github/workflows/pull-request-labeler.yml (hunk 2 above)
Comment (snazy) [reply .github/workflows/pull-request-labeler.yml:29]: Was thinking about it, but what it you open a PR, add the label yourself, and just want to make sure that things you changed elsewhere still works.

File: .github/workflows/pull-request-labeler.yml (hunk 3 above)
Comment (snazy) [reply .github/workflows/pull-request-labeler.yml:18]: Just style from everywhere else.

**Comment by snazy** [comment]
Side note: also thinking about making the "optional" checks mandatory. But there's no good solution from GH. If a check is mandatory but not executed ("Skipped"), then the "required check" will fail. GH's workaround is to add another WF with the same name, that runs, but does nothing, and has its filter configured in the "opposite way". https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+pr-helm:
+- 'helm/**'
+- '.github/**'
+
+pr-native:
+- 'gradle/wrapper/**'
+- 'servers/**'
+- '.github/**'
+
+pr-integrations:
```
Comment (dimas-b) [inline .github/labeler.yml:29]: We should probably add `pr-integrations` also in case of `model` changes, WDYT?

Comment (snazy) [reply .github/labeler.yml:29]: Sure, mind opening a PR?

Comment (dimas-b) [reply .github/labeler.yml:29]: #6132

**Pull request merged**